shared/middleware/helpers/storage_apps.rb

```diff
@@ -204,6 +214,27 @@ def reset_abuse(channel_id)
     0
   end
 
+  #
+  # Disables or enables automated content moderation for this project by
+  # altering the value for skip_content_moderation.
+  # @param [String] channel_id - an encrypted channel id
+  # @param [Boolean] should_skip, whether the content moderation should be
+  # skipped or not for this project.
+  # @raise [NotFound] if the channel does not exist or already has the desired
+  # value for skip_content_moderation
+  #
+  def change_content_moderation(channel_id, should_skip)
+    _owner, id = storage_decrypt_channel_id(channel_id)
+
+    row = @table.where(id: id).exclude(state: 'deleted').first
+    raise NotFound, "channel `#{channel_id}` not found" unless row
+
+    update_content_moderation = @table.where(id: id).exclude(state: 'deleted').update({skip_content_moderation: should_skip})
```

Comment:
Because `update` is safe if the project is not present and it returns the number of rows modified, you can implement this API with one query instead of two.

```rb
_owner, id = storage_decrypt_channel_id(channel_id)
rows_changed = @table.
  where(id: id).
  exclude(state: 'deleted').
  update({skip_content_moderation: should_skip})
raise NotFound, "channel `#{channel_id}` not found" unless rows_changed > 0
```